camel-route-support/src/main/java/io/kaoto/backend/model/deployment/camelroute/IntegrationSpec.java

```diff
@@ -1,41 +1,53 @@
 package io.kaoto.backend.model.deployment.camelroute;
 
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import io.fabric8.kubernetes.api.model.KubernetesResource;
 import io.kaoto.backend.model.deployment.kamelet.Flow;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import org.apache.camel.v1.integrationspec.Flows;
 
-import java.io.Serializable;
+import java.util.LinkedList;
 import java.util.List;
 
-/**
- * 🐱miniclass IntegrationSpec (Integration)
- */
-
-@JsonPropertyOrder({"flows"})
-@JsonDeserialize(
-        using = JsonDeserializer.None.class
-)
-@JsonIgnoreProperties(ignoreUnknown = true)
-public class IntegrationSpec implements KubernetesResource, Serializable {
-
-    /*
-     * 🐱property flows: Flow[]
-     *
-     * List of flows this route supports.
+@JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"configuration","dependencies","flows","integrationKit","profile","replicas","repositories",
+        "serviceAccountName","sources","template","traits"})
+@JsonDeserialize(using = JsonDeserializer.None.class)
+@RegisterForReflection
+public class IntegrationSpec extends org.apache.camel.v1.IntegrationSpec {
+    /**
+     * a source in YAML DSL language which contain the routes to run
      */
     @JsonProperty("flows")
-    private List<Flow> flows;
+    @JsonPropertyDescription("a source in YAML DSL language which contain the routes to run")
+    @JsonSetter(nulls = Nulls.SKIP)
+    private List<Flow> flows = new LinkedList<>();
 
-    public List<Flow> getFlows() {
+    public List<Flow> get_flows() {
         return flows;
     }
 
-    public void setFlows(
-            final List<Flow> flows) {
+    public void set_flows(List<Flow> flows) {
         this.flows = flows;
     }
+
+    @Override
+    @Deprecated
+    @JsonIgnore
+    public List<Flows> getFlows() {
+        return super.getFlows();
```

Comment:
Because I needed to add something to the function content. It could return a null, but just in case someone wants to use it to check something, even if it is marked as deprecated.